dashboard/app/models/concerns/text_to_speech.rb

```diff
@@ -118,9 +118,11 @@ def self.sanitize(text)
     TTSSafeRenderer.render(text)
   end
 
-  def tts_upload_to_s3(text, context = nil)
-    filename = tts_path(text)
-    TextToSpeech.tts_upload_to_s3(text, filename, context)
+  def tts_upload_to_s3(text, context = nil, locale: I18n.locale)
```

Comment:
Can you speak a bit as to why you decided to update the underlying tts implementation to provide language-switching functionality, rather than continuing to rely on the caller to define the current language? I don't think it's a bad change, but it does represent a difference in the way we're thinking about this interface.

Reply:
I wouldn't guess that this method depends on i18n locale without digging deeply into it.

So I decided to add this `locale` argument to provide more context to the method and make it more clear.

I'll change it back if you want

Reply:
It does definitely make it more clear! I'm not opposed to this change, I just want to make sure we're applying it consistently. If there is anywhere else in the codebase that was using the old approach of setting the locale before calling these methods, those should also be updated to use the new approach. That way we can look at the effects of this change on the codebase as a whole when deciding whether to update or stay with what we have!

Might be worth separating that work out into a different PR

Reply:
We have only one place where we use it, and I'm going to correct this in the next PR for the task [P20-554: Create unit tests for TTS I18n Static Messages sync-out](https://codedotorg.atlassian.net/browse/P20-554)
https://github.com/code-dot-org/code-dot-org/blob/94426f3a40be6e701be9f92258471c7c8cdc68ec/dashboard/scripts/update_tts_i18n_static_messages.rb#L114